module/Finna/src/Finna/RecordDriver/SolrLido.php

```diff
@@ -2115,8 +2130,38 @@ public function getPhysicalLocations(): array
                     $part = $part->partOfPlace;
                 }
             }
+            $placeId = $repository->repositoryLocation->placeID ?? [];
             if ($locations) {
-                $results[] = implode(', ', $locations);
+                $location = implode(', ', $locations);
+                if ($placeId) {
+                    $attr = $placeId->attributes();
+                    $idTypeFirst = trim((string)($attr->type));
+                    $result['type'] = $idTypeFirst;
+                    $result['id'] = $idTypeFirst ? "($idTypeFirst)$placeId" : $placeId;
+                    foreach ($placeId as $idItem) {
```

Comment:
Nää voi kattoa kaikki samassa loopissa ja esim $i = 0; kattoa onko ensimmäinen mistä otetaan arvot.

foreach ($repository->repositoryLocation->placeID ?? [] as $placeID)

if ($i++ = 0) { 
 // Talleta eka id ja type omiinsa.
}

Reply:
Mietin et miksköhän tosta ekasta otetaan noi idtyypit ja id kun se luottaa sitten metadatan järjestykseen mikä ei oo validi. Pitäis olla ennemmin joku tunniste minkä mukaan se otetaan.